lib/membrane_tcp/sink.ex

```diff
@@ -8,27 +8,15 @@ defmodule Membrane.TCP.Sink do
   alias Membrane.TCP.{CommonSocketBehaviour, Socket}
 
   def_options connection_side: [
-                spec: :client | :server,
-                default: :server,
+                spec:
+                  :server
+                  | :client
+                  | {:client, server_address :: :inet.ip_address(),
+                     server_port_no :: :inet.port_number()},
                 description: """
-                Determines whether this element will behave like a server or a client when
-                establishing TCP connection.
-                """
-              ],
-              server_address: [
-                spec: :inet.ip_address() | nil,
-                default: nil,
-                description: """
-                An IP Address of the server the packets will be sent to.
-                (nil in case of `connection_side: :server`)
-                """
-              ],
-              server_port_no: [
-                spec: :inet.port_number() | nil,
-                default: nil,
-                description: """
-                A TCP port number of the server the packets will be sent to.
-                (nil in case of `connection_side: :server`)
+                Determines whether this element will operate like a server or a client when
+                establishing TCP connection. In case of client-side connection server address
+                and port number are required, unless `local_socket` is provided.
```

Comment:
As far as I understand, some information is duplicated between this option and `local_socket`, so I'd suggest that `local_socket` could just accept `:gen_tcp.socket`